src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Extensions/ResolveFieldContextExtensions.cs

```diff
@@ -8,14 +8,15 @@ namespace VirtoCommerce.XDigitalCatalog.Extensions
 {
     public static class ResolveFieldContextExtensions
     {
-        public static T GetCatalogQuery<T>(this IResolveFieldContext context) where T: ICatalogQuery
+        public static T GetCatalogQuery<T>(this IResolveFieldContext context) where T : ICatalogQuery
         {
             var result = AbstractTypeFactory<T>.TryCreateInstance();
             result.StoreId = context.GetArgument<string>("storeId") ?? context.GetValue<string>("storeId");
-            result.UserId = context.GetArgument<string>("userId") ?? context.GetValue<string>("userId");
+            result.UserId = context.GetArgument<string>("userId") ?? context.GetValue<string>("userId") ?? context.GetCurrentUserId();
```

Comment:
Could we also move to some extension method `GetArgumentOrValue` - as it seems it uses in many places 
```
context.GetArgument<string>("storeId") ?? context.GetValue<string>("storeId")
```

Reply:
added